src/sound_openal.cpp

```diff
@@ -175,14 +164,104 @@ SoundBuffer* loadOggFile(const std::string &filepath)
 				<<"preparing sound buffer"<<std::endl;
 	}
 
-	infostream<<"Audio file "<<filepath<<" loaded"<<std::endl;
+	infostream << "Audio file "
+		<< filename_for_logging << " loaded" << std::endl;
 
 	// Clean up!
-	ov_clear(&oggFile);
+	ov_clear(oggFile);
 
 	return snd;
 }
 
+SoundBuffer *load_ogg_from_file(const std::string &path)
+{
+	OggVorbis_File oggFile;
+
+	// Try opening the given file.
+	// This requires libvorbis >= 1.3.2, as
+	// previous versions expect a non-const char *
+	if (ov_fopen(path.c_str(), &oggFile) != 0) {
+		infostream << "Audio: Error opening " << path
+			<< " for decoding" << std::endl;
+		return NULL;
+	}
+
+	return load_opened_ogg_file(&oggFile, path);
+}
+
+struct buffer_sound_t {
```

Comment:
`BufferSound`?  Also, below, perhaps `buffer_sound` or at least `data_source` would be more appropriate.